### PR TITLE
fix: don't throw ipfs errors, use defaults

### LINKED ISSUE #146: Unsupported proposal metadata preventing application use
Originally reported by @S4mmyb in slack.

If a user submits a group proposal using the CLI and the metadata for the proposal is not a valid JSON string, the group page and the proposal page is inaccessible to the user.

A user should be able to continue using the application regardless of how the metadata within a proposal is formatted (i.e. we should have fallbacks in place). The proposal metadata should also support IPFS (same as group and policy metadata).

This occurred as a result of a user submitting the following proposal:

```
{
 "group_policy_address": "regen1ys0dhh6x5s55h2g37zrnc7kh630jfq5p77as8pwyn60ax9zzqh9q4reeah",
 "messages": [
  {
   "@type": "/cosmos.group.v1.MsgCreateGroupPolicy",
   "admin": "regen1ys0dhh6x5s55h2g37zrnc7kh630jfq5p77as8pwyn60ax9zzqh9q4reeah",
   "group_id": "6",
   "metadata": "{\"name\": \"Biochar Carbon Credit Class Buffer Pool Account\"}",
   "decision_policy": {
        "@type": "/cosmos.group.v1.PercentageDecisionPolicy",
        "percentage": "1",
        "windows": {
            "voting_period": "120h",
            "min_execution_period": "0s"
        }
    }
  }
 ],
 "metadata": "ipfs://CID",
 "proposers": ["regen1v2ncquer9r2ytlkxh2djmmsq3e8we6rjc9snfn"]
}
```

Notice that the proposal metadata is `ipfs://CID`. This prevents the following group page from being accessible:

- https://groups.regen.network/6

### PR DESCRIPTION
**Pull request by wgwz**
Closes: #146 

Testing:

1. create a proposal with an invalid ipfs url for a group policy address i.e.

```
➜  ~ cat proposal.json 
{
  "@type": "/cosmos.group.v1.MsgSubmitProposal",
  "group_policy_address": "regen1mqevjln5hxv3qgd3c4m5zjeeand5hkc7r33ty82fjukw9shxjh6s6d4umv",
  "proposers": ["regen1fhjpuh6qh6kjsedtmfxrayvmzlknkzde92ky2j"],
  "metadata": "ipfs://CID",
  "messages": [],
  "exec": "EXEC_UNSPECIFIED"
}
```

2. submit the proposal via cli i.e.

```
➜  ~ ~/go/bin/regen tx group submit-proposal proposal.json --keyring-backend test --chain-id test 
```

3. Go to the page for the group in the groups-ui, observe that the group page and proposal page loads successfully.

4. Checkout `dev`, repeat 1-3, observe the failure.

### CODE REVIEW

**Review by ryanchristo — COMMENTED** [review]
You can also test using the deploy preview connected to mainnet [here](https://deploy-preview-149--regen-groups-ui.netlify.app/6).

Any ideas why the proposal is not showing up when we hit this error? I think we want to have a fallback in place and show the proposal. It looks like this solves the immediate issue though with the page not loading.

I'm ok with addressing the fallback separately if not a quick fix.

Screenshot from 2023-09-12 15-17-39

**Comment by wgwz** [comment]
@ryanchristo it was showing up for me when i tested locally, i’ll take a better look but it may just be a separate issue. i’m kind of confused by what’s showing in your logs, unless i’m mistaken i thought the new log statement i added would have shown up there.

oh you know what, maybe it was because i tested the metadata as precisely ipfs://CID whereas the metadata in the proposal Sam added might have had quote characters around it. i’ll look at more and see if that was it, and if so i’ll just do a fix here

**Comment by ryanchristo** [comment]
You should also be able to test locally with the proposal on mainnet: http://localhost:5173/6

**Comment by ryanchristo** [comment]
> You should also be able to test locally with the proposal on mainnet: http://localhost:5173/6

Actually you would hit a CORS error if you did this with the production indexer so nvm.

**Comment by wgwz** [comment]
~~> Any ideas why the proposal is not showing up when we hit this error? I think we want to have a fallback in place and show the proposal. It looks like this solves the immediate issue though with the page not loading.~~

~~@ryanchristo i think the reason that proposal doesn't show up is because the deploy preview is not connected to the production indexer either, right? the groups-ui deploy previews are connected to the staging indexer only, unless i'm just totally misremembering stuff.~~

nevermind. that's not the case. duh

**Comment by wgwz** [comment]
> Actually you would hit a CORS error if you did this with the production indexer so nvm.

we can disable CORS temporarily on the server, and that should make it work, and allow better debugging

**Comment by wgwz** [comment]
it looks like a possible explanation for why the proposals are not showing up here:
https://deploy-preview-149--regen-groups-ui.netlify.app/6
is a bug in the historical proposals fetching.
it looks like this group has two group policies.
TL;DR there may be a bug in the `useGroupHistoricalProposals` function in this case.
i'm not sure how well i tested this scenario.
it's possible that because one of the indexer API calls returns an empty result set, that we are using that.
for one of the policy addresses we find proposals and for the other policy we get the empty set.
you can see this in the network tab when you have the group page open, and requests filtered to "graphql":

Request/response for first policy:
Screen Shot 2023-09-12 at 7 25 29 PM
Screen Shot 2023-09-12 at 7 25 36 PM

Request/response for second policy:
Screen Shot 2023-09-12 at 7 25 43 PM
Screen Shot 2023-09-12 at 7 25 49 PM

it's fairly obvious that's a bug because we get some historical proposals, so those should be showing.
pretty sure this explains it

**Review by ryanchristo — APPROVED** [review]
tACK. Sounds like a separate issue. This pr looks good to me.

**Comment by wgwz** [comment]
Opened an issue for the missing proposals:
https://github.com/regen-network/groups-ui/issues/150
